wrangler-proto/src/main/java/io/cdap/wrangler/proto/workspace/v2/SampleSpec.java

```diff
@@ -29,13 +33,28 @@ public class SampleSpec {
   private final String connectionType;
   private final String path;
   private final Set<StageSpec> relatedPlugins;
+  // This gives the different methods the current connection supports for sampling
+  // (e.g. first N rows, randomized, stratified)
+  private final Set<SampleType> supportedSampleTypes;
+  private SampleRequest sampleRequest;
 
   public SampleSpec(String connectionName, String connectionType, @Nullable String path,
-                    Set<StageSpec> relatedPlugins) {
+                    Set<StageSpec> relatedPlugins, Set<SampleType> supportedSampleTypes, SampleRequest sampleRequest) {
     this.connectionName = connectionName;
     this.connectionType = connectionType;
     this.path = path;
     this.relatedPlugins = relatedPlugins;
+    if (supportedSampleTypes != null) {
+      this.supportedSampleTypes = supportedSampleTypes;
+    } else {
+      this.supportedSampleTypes = new HashSet<>();
+    }
+    this.sampleRequest = sampleRequest;
+  }
+
+  public SampleSpec(String connectionName, String connectionType, @Nullable String path,
+                    Set<StageSpec> relatedPlugins) {
+    this(connectionName, connectionType, path, relatedPlugins, null, null);
```

Comment:
pass in `Collections.emptySet` instead of null for Collections

Reply:
Thanks, fixed!